packages/daemon/src/context/ContextMutations.js

```diff
@@ -208,6 +208,24 @@ export default class ContextMutations {
     return user
   }
 
+  async updateUser(
+    userID: ID,
+    profile: { name?: ?string, avatar?: ?string },
+  ): Promise<void> {
+    const user = this._context.openVault.identities.getOwnUser(userID)
+    if (!user) throw new Error('User not found')
+
+    user.profile = {
+      name: profile.name || user.profile.name,
+      avatar: profile.avatar || user.profile.avatar,
+    }
+    await this._context.openVault.save()
```

Comment:
Shouldn't this mutation publish the `user_changed` event and have the vault saving and feed update be handled by the side-effect logic rather than performing these side-effects here?

Reply:
Yeah but that work is being done in a different branch so I didn't want to mix concerns